fuse-products/src/main/java/software/tnb/product/application/App.java

```diff
@@ -103,8 +105,12 @@ public Path getLogPath() {
 
     public void waitUntilReady() {
         if (shouldRun()) {
+            Supplier<FailureCauseException> failureSupplier = null;
+            if (!TestConfiguration.streamLogs()) {
+                failureSupplier = () -> new FailureCauseException("The Camel app failed to start.", getLog().toString());
+            }
             WaitUtils.waitFor(() -> isReady() && isCamelStarted(), this::isFailed, 1000L,
-                "Waiting until the integration " + getName() + " is running");
+                "Waiting until the integration " + getName() + " is running", failureSupplier);
```

Comment:
`, TestConfiguration.streamLogs() ? null : () -> new FailureCauseException("The Camel app failed to start.", getLog().toString())` does not work?

Reply:
of course it does, and you're right, looks much better